src/pages/Home/ReceiveModal/__tests__/ReceiveModal.test.tsx

```diff
@@ -24,4 +24,35 @@ describe("ReceiveModal", () => {
 
     expect(await screen.findByRole("img")).toBeDefined();
   });
+
+  test("Retrieves a new address upon clicking the refresh button", async () => {
+    const user = userEvent.setup();
+    render(<ReceiveModal onClose={() => {}} />);
+
+    const onChainBtn = screen.getByRole("button", { name: "wallet.on_chain" });
+
+    await user.click(onChainBtn);
+
+    // one-time address override
+    const newMockAddress = "muZ3nH1uH2U5JBhZRvYU46MdChbCifSghQ";
+    server.use(
+      http.post(
+        "/api/lightning/new-address",
+        () => {
+          return new HttpResponse(newMockAddress, {
+            status: 200,
+          });
+        },
+        { once: true },
+      ),
+    );
+
+    const refreshBtn = await screen.findByRole("button", {
```

Comment:
Looks like the test is failing - could you check that please? :)

Reply:
Hey @cstenglein , no worries at all. Happy to contribute, and hopefully there'll be a chance to keep helping out.

Interesting, I'm checking this one out. I noticed that there's an uncaught request going out, and I managed to catch it with a `beforeEach` instead of `beforeAll` at the beginning of these tests. However, it seems like the error is an "unknown login error" which does not seem to be happening when running the tests locally. Can you think of any reasons why that could be?

In any case, let's maybe see if this takes care of it.

Reply:
Well, it seems to have done the trick, right?

Let me know if there's anything else and I'll check it out. 🙂

Reply:
Yes, the reason is the `useEffect` hook in the `ReceiveOnChain` component, which fetches the new address at the start. 

Looking good, merging, thank you!